sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs

```diff
@@ -0,0 +1,434 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using AccountSetting = System.Collections.Generic.KeyValuePair<string, System.Func<string, bool>>;
+using ConnectionStringFilter = System.Func<System.Collections.Generic.IDictionary<string, string>, System.Collections.Generic.IDictionary<string, string>>;
+
+namespace Azure.Data.Tables
+{
+    internal class TableConnectionString
+    {
+        /// <summary>
+        /// Gets the endpoints for the Table service at the primary and secondary location, as configured for the storage account.
+        /// </summary>
+        /// <value>A <see cref="System.Uri"/> containing the Table service endpoints.</value>
+        public (Uri PrimaryUri, Uri SecondaryUri) TableStorageUri { get; set; }
+
+        /// <summary>
+        /// Gets the credentials used to create this <see cref="TableConnectionString"/> object.
+        /// </summary>
+        /// <value>A StorageCredentials object.</value>
+        public object Credentials { get; set; }
```

Comment:
AFAIK this can only be a shared key credential

Reply:
It can also be the SAS token string